src/main/java/seedu/address/model/contact/TsContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,49 @@
+package seedu.address.model.contact;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.StringUtil;
+import seedu.address.commons.util.ToStringBuilder;
+
+/**
+ * Tests that a {@code Person}'s {@code TechStack} matches any of the keywords given.
+ */
+public class TsContainsKeywordsPredicate implements Predicate<Contact> {
+    private final List<String> techKeywords;
+
+    public TsContainsKeywordsPredicate(List<String> techKeywords) {
+        this.techKeywords = techKeywords;
+    }
+
+    @Override
+    public boolean test(Contact contact) {
+        return techKeywords.stream()
+                .allMatch(keyword ->
+                        contact.getTechStack().stream()
+                                .anyMatch(ts ->
+                                        StringUtil.containsWordIgnoreCase(ts.techStackName, keyword)
+                                )
+                );
+    }
```

Comment:
This could be formatted better

```suggestion
    @Override
    public boolean test(Contact contact) {
        return techKeywords.stream().allMatch(keyword -> 
            contact.getTechStack().stream().anyMatch(ts ->
                StringUtil.containsWordIgnoreCase(ts.techStackName, keyword)
            )
        );
    }
```